holoviews/core/spaces.py

```diff
@@ -454,6 +463,12 @@ class DynamicMap(HoloMap):
        """)
 
     def __init__(self, callback, initial_items=None, **params):
+
+        # Set source to self if not already specified
+        for stream in params.get('streams',[]):
```

Comment:
This should happen after the parameters have been instantiated. I wouldn't recommend it but someone could set a number of default streams directly on DynamicMap.